nullplatform/resource_notification_channel.go

```diff
@@ -203,9 +203,34 @@ func NotificationChannelCreate(d *schema.ResourceData, m any) error {
 	case "agent":
 		if agent, ok := config["agent"].([]interface{}); ok && len(agent) > 0 {
 			agentMap := agent[0].(map[string]any)
-			flatConfig["api_key"] = agentMap["api_key"]
 			if command, ok := agentMap["command"].([]any); ok {
-				flatConfig["command"] = command[0]
+				if dataMap, ok := command[0].(map[string]any)["data"].(map[string]any); ok {
+					dataConfig := map[string]any{
+						"cmdline": dataMap["cmdline"],
+					}
+
+					if dataMap["arguments"] != nil {
+						var argumentsList []string
+						if err := json.Unmarshal([]byte(dataMap["arguments"].(string)), &argumentsList); err != nil {
```

Comment:
aca no hace falta el assertion, porque string por abajo es una tira de bytes, lo que cambias es el criterio que usas al leer esos bytes

https://stackoverflow.com/questions/20494229/what-is-the-difference-between-type-conversion-and-type-assertion

Reply:
does not apply anymore